src/Game.js

```diff
@@ -10,17 +12,26 @@ class Game {
     this.legalMoves = [];
     this.selectedPiece = null;
     this.board.gameAreaHandler.addEventListener('click', e => this.onClick(e));
+
   }
 
   onClick(e) {
     const element = e.target.classList.contains('square') ? e.target : e.target.parentElement;
-    if (this.legalMoves.length !== 0) {
-      this.handleMove(element);
+
+    const { id } = element;
+    if (this.possibleMoves.length !== 0) {
+      if (this.possibleMoves.includes(id)) this.handleMove(element);
+      else this.removeSelection();
```

Comment:
Żeby trochę ujednolicić nazewnictwo zmieniłem this. possibleMoves na this.legalMoves. Tutaj właśnie powinnaś odwołać się do this.legalMoves.